pkg/workflow/controller/handlers/workflowrun/handler.go

```diff
@@ -172,7 +185,15 @@ func (h *Handler) ObjectDeleted(obj interface{}) {
 // * notification endpoint is configured
 // * without notification sent label
 func (h *Handler) sendNotification(wfr *v1alpha1.WorkflowRun) error {
-	if meta.LabelExists(wfr.Labels, meta.LabelWorkflowRunNotificationSent) {
+	h.notifyLock.Lock(wfr.Name)
+	defer h.notifyLock.Unlock(wfr.Name)
```

Comment:
Seems no need to use lock, controller handles event in a single thread one by one.

Reply:
Have fixed, PTAL @cd1989